src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -234,6 +272,29 @@ internal InvocationVersionOne CreateInvocation(Invocation v2Invocation)
             return invocation;
         }
 
+        internal LocationVersionOne CreateLocation(Location v2Location)
+        {
+            LocationVersionOne location = null;
+
+            if (v2Location != null)
+            {
+                location = new LocationVersionOne
+                {
+                    FullyQualifiedLogicalName = v2Location.FullyQualifiedLogicalName,
+                    Properties = v2Location.Properties,
+                    ResultFile = CreatePhysicalLocation(v2Location.PhysicalLocation)
+                };
+
+                if (!string.IsNullOrWhiteSpace(v2Location.FullyQualifiedLogicalName) &&
+                    _currentV2Run.LogicalLocations?.ContainsKey(v2Location.FullyQualifiedLogicalName) == true)
```

Comment:
If your v2 location has an FQLN, and the logical locations dictionary does not yet have an entry for that FQLN, then you must create one. Otherwise you will lose the decorated name.

Reply:
Not following. I'm getting the decorated name *from* the logical location.

Reply:
Oh, I see. _If_ there is a v2 logical location object that contains the decorated name, then copy it to the v1 location object. Got it.

Reply:
Yeah, reversed what we are doing v1 -> v2